File: multiple-languages/go/alicloudroscdkga/IBandwidthPackageAcceleratorAddition.go
package alicloudroscdkga

import (
	_jsii_ "github.com/aws/jsii-runtime-go/runtime"

	"github.com/alibabacloud-go/ros-cdk/alicloudroscdkcore"
	"github.com/alibabacloud-go/ros-cdk/alicloudroscdkga/internal"
)

// Represents a `BandwidthPackageAcceleratorAddition`.
type IBandwidthPackageAcceleratorAddition interface {
	alicloudroscdkcore.IResource
	// Attribute AcceleratorId: The ID of the Global Accelerator instance.
	AttrAcceleratorId() interface{}
	// Attribute BandwidthPackageId: The ID of the bandwidth package which is associated.
	AttrBandwidthPackageId() interface{}
	Props() *BandwidthPackageAcceleratorAdditionProps
}

// The jsii proxy for IBandwidthPackageAcceleratorAddition
type jsiiProxy_IBandwidthPackageAcceleratorAddition struct {
	internal.Type__alicloudroscdkcoreIResource
}

func (j *jsiiProxy_IBandwidthPackageAcceleratorAddition) AttrAcceleratorId() interface{} {
	var returns interface{}
	_jsii_.Get(
		j,
		"attrAcceleratorId",
		&returns,
	)
	return returns
}

func (j *jsiiProxy_IBandwidthPackageAcceleratorAddition) AttrBandwidthPackageId() interface{} {
	var returns interface{}
	_jsii_.Get(
		j,
		"attrBandwidthPackageId",
		&returns,
	)
	return returns
}

func (j *jsiiProxy_IBandwidthPackageAcceleratorAddition) Props() *BandwidthPackageAcceleratorAdditionProps {
	var returns *BandwidthPackageAcceleratorAdditionProps
	_jsii_.Get(
		j,
		"props",
		&returns,
	)
	return returns
}

File: multiple-languages/dotnet/AlibabaCloud.SDK.ROS.CDK.Ga/AlibabaCloud/SDK/ROS/CDK/Ga/IBandwidthPackageAcceleratorAddition.cs
using Amazon.JSII.Runtime.Deputy;

#pragma warning disable CS0672,CS0809,CS1591

namespace AlibabaCloud.SDK.ROS.CDK.Ga
{
    /// <summary>Represents a `BandwidthPackageAcceleratorAddition`.</summary>
    [JsiiInterface(nativeType: typeof(IBandwidthPackageAcceleratorAddition), fullyQualifiedName: "@alicloud/ros-cdk-ga.IBandwidthPackageAcceleratorAddition")]
    public interface IBandwidthPackageAcceleratorAddition : AlibabaCloud.SDK.ROS.CDK.Core.IResource
    {
        /// <summary>Attribute AcceleratorId: The ID of the Global Accelerator instance.</summary>
        [JsiiProperty(name: "attrAcceleratorId", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
        object AttrAcceleratorId
        {
            get;
        }

        /// <summary>Attribute BandwidthPackageId: The ID of the bandwidth package which is associated.</summary>
        [JsiiProperty(name: "attrBandwidthPackageId", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
        object AttrBandwidthPackageId
        {
            get;
        }

        [JsiiProperty(name: "props", typeJson: "{\"fqn\":\"@alicloud/ros-cdk-ga.BandwidthPackageAcceleratorAdditionProps\"}")]
        AlibabaCloud.SDK.ROS.CDK.Ga.IBandwidthPackageAcceleratorAdditionProps Props
        {
            get;
        }

        /// <summary>Represents a `BandwidthPackageAcceleratorAddition`.</summary>
        [JsiiTypeProxy(nativeType: typeof(IBandwidthPackageAcceleratorAddition), fullyQualifiedName: "@alicloud/ros-cdk-ga.IBandwidthPackageAcceleratorAddition")]
        internal sealed class _Proxy : DeputyBase, AlibabaCloud.SDK.ROS.CDK.Ga.IBandwidthPackageAcceleratorAddition
        {
            private _Proxy(ByRefValue reference): base(reference)
            {
            }

            /// <summary>Attribute AcceleratorId: The ID of the Global Accelerator instance.</summary>
            [JsiiProperty(name: "attrAcceleratorId", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
            public object AttrAcceleratorId
            {
                get => GetInstanceProperty<object>()!;
            }

            /// <summary>Attribute BandwidthPackageId: The ID of the bandwidth package which is associated.</summary>
            [JsiiProperty(name: "attrBandwidthPackageId", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
            public object AttrBandwidthPackageId
            {
                get => GetInstanceProperty<object>()!;
            }

            [JsiiProperty(name: "props", typeJson: "{\"fqn\":\"@alicloud/ros-cdk-ga.BandwidthPackageAcceleratorAdditionProps\"}")]
            public AlibabaCloud.SDK.ROS.CDK.Ga.IBandwidthPackageAcceleratorAdditionProps Props
            {
                get => GetInstanceProperty<AlibabaCloud.SDK.ROS.CDK.Ga.IBandwidthPackageAcceleratorAdditionProps>()!;
            }

            /// <summary>The environment this resource belongs to.</summary>
            /// <remarks>
            /// For resources that are created and managed by the CDK
            /// (generally, those created by creating new class instances like Role, Bucket, etc.),
            /// this is always the same as the environment of the stack they belong to;
            /// however, for imported resources
            /// (those obtained from static methods like fromRoleArn, fromBucketName, etc.),
            /// that might be different than the stack they were imported into.
            /// </remarks>
            [JsiiProperty(name: "env", typeJson: "{\"fqn\":\"@alicloud/ros-cdk-core.ResourceEnvironment\"}")]
            public AlibabaCloud.SDK.ROS.CDK.Core.IResourceEnvironment Env
            {
                get => GetInstanceProperty<AlibabaCloud.SDK.ROS.CDK.Core.IResourceEnvironment>()!;
            }

            /// <summary>The stack in which this resource is defined.</summary>
            [JsiiProperty(name: "stack", typeJson: "{\"fqn\":\"@alicloud/ros-cdk-core.Stack\"}")]
            public AlibabaCloud.SDK.ROS.CDK.Core.Stack Stack
            {
                get => GetInstanceProperty<AlibabaCloud.SDK.ROS.CDK.Core.Stack>()!;
            }

            /// <summary>The construct tree node for this construct.</summary>
            [JsiiProperty(name: "node", typeJson: "{\"fqn\":\"@alicloud/ros-cdk-core.ConstructNode\"}")]
            public AlibabaCloud.SDK.ROS.CDK.Core.ConstructNode Node
            {
                get => GetInstanceProperty<AlibabaCloud.SDK.ROS.CDK.Core.ConstructNode>()!;
            }
        }
    }
}
